### PR TITLE
Add support for ERC-20 custom tokens

### LINKED ISSUE #83: Add support for ERC-20 custom tokens
## Rationale

This feature will add a new text input on top of the current Token list to enable the filtering of the existing tokens and pasting custom ERC-20 token contract addresses to bridge custom tokens.

Once the user has pasted the address of a contract in the input, as long as the contract exists, she can do a bridge (provided the account has funds) and click the "Import" button to save the token in the local storage for further use in the future. Imported tokens can later be removed.

### PR DESCRIPTION
**Pull request by amonsosanz**
Closes #83 

### What does this PR does?

This PR adds a new text input on top of the current Token list to:

- Enable the filtering of the existing tokens.
- Pasting custom ERC-20 token contract addresses to bridge custom tokens.

Once the user has pasted the address of a contract in the input, as long as the contract exists, she can do a bridge (provided the account has funds) and click the "Import" button to save the token in the local storage for further use in the future. Imported tokens can later be removed.

### CODE REVIEW

File: src/contexts/bridge.context.tsx [hunk 1]
```diff
@@ -14,13 +15,19 @@ import { Bridge__factory } from "src/types/contracts/bridge";
 import { Erc20__factory } from "src/types/contracts/erc-20";
 import { BRIDGE_CALL_GAS_INCREASE_PERCENTAGE } from "src/constants";
 import { calculateFee } from "src/utils/fees";
+import tokenIconDefaultUrl from "src/assets/icons/tokens/erc20-icon.svg";
```
Comment (amonsosanz) [inline src/contexts/bridge.context.tsx:18]: We have to make sure that this URL is properly resolved once deployed.

File: src/views/home/components/token-list/token-list.styles.ts [hunk 2]
```diff
@@ -14,19 +14,19 @@ const useListStyles = createUseStyles((theme: Theme) => ({
   },
   card: {
     width: "100%",
-    maxWidth: 426,
-    padding: theme.spacing(2),
-  },
-  search: {
+    maxWidth: 500,
```
Comment (adrifdez) [inline src/views/home/components/token-list/token-list.styles.ts:17]: Why extend it to 500?

File: src/adapters/storage.ts [hunk 3]
```diff
@@ -15,6 +18,50 @@ export function setCurrency(currency: Currency): void {
...
+}
+
+export function setCustomTokens(tokens: Token[]): Token[] {
+  return setStorageByKey({
+    key: CUSTOM_TOKENS_KEY,
+    value: tokens,
+  });
+}
+
+export function addCustomToken(token: Token): Token[] {
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/adapters/storage.ts:51]: The storage API doesn't care about the specific implementation of the `TokenList` component and always returns the updated version of the storage after each operation. If a client of the storage API does not require the updated version of the storage, can safely ignore it.

File: src/adapters/storage.ts [hunk 4]
```diff
@@ -15,6 +18,50 @@ export function setCurrency(currency: Currency): void {
...
+  const isAlreadyAdded = customTokens.find((tkn) => tkn.address === token.address);
+  if (isAlreadyAdded) {
+    return customTokens;
+  } else {
+    return setCustomTokens([token, ...customTokens]);
+  }
+}
+
+export function removeCustomToken(token: Token): Token[] {
+  return setCustomTokens(getCustomTokens().filter((tkn) => tkn.address !== token.address));
```
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/adapters/storage.ts:62]: [Same as before](https://github.com/hermeznetwork/bridge-ui/pull/86#discussion_r877265839).

File: src/views/home/components/token-list/token-list.styles.ts (hunk 2 above)
Comment (amonsosanz) [reply src/views/home/components/token-list/token-list.styles.ts:17]: To make sure all addresses fit and give some extra room till the `x` blue cross:

long-address

File: src/contexts/bridge.context.tsx (hunk 1 above)
Comment (amonsosanz) [reply src/contexts/bridge.context.tsx:19]: I have `npm run build && npm run start:prod` and confirmed that the file is properly served from `http://localhost/static/media/erc20-icon.dd72ab1b.svg`

**Review by adrifdez — APPROVED** [review]

**Pull request merged**